web/secondary_toolbar.js

```diff
@@ -29,173 +29,228 @@
 var SCROLLBAR_PADDING = uiUtils.SCROLLBAR_PADDING;
 var mozL10n = uiUtils.mozL10n;
 
-var SecondaryToolbar = {
-  opened: false,
-  previousContainerHeight: null,
-  newContainerHeight: null,
+/**
+ * @typedef {Object} SecondaryToolbarOptions
+ * @property {HTMLDivElement} toolbar - Container for the secondary toolbar.
+ * @property {HTMLButtonElement} toggleButton - Button for toggling the
+ *   visibility of the secondary toolbar.
+ * @property {HTMLButtonElement} presentationModeButton - Button for toggling
+ *   presentation mode.
+ * @property {HTMLButtonElement} openFile - Button for opening a file.
+ * @property {HTMLButtonElement} print - Button for printing the document.
+ * @property {HTMLButtonElement} download - Button for downloading the document.
+ * @property {HTMLLinkElement} viewBookmark - Button for obtaining a bookmark
+ *   link to the current location in the document.
+ * @property {HTMLButtonElement} firstPage - Button for going to the first page
+ *   in the document.
+ * @property {HTMLButtonElement} lastPage - Button for going to the last page
+ *   in the document.
+ * @property {HTMLButtonElement} pageRotateCw - Button for rotating the page
+ *   clockwise.
+ * @property {HTMLButtonElement} pageRotateCcw - Button for rotating the page
+ *   counterclockwise.
+ * @property {HTMLButtonElement} toggleHandTool - Button for toggling the hand
+ *   tool.
+ * @property {HTMLButtonElement} documentPropertiesButton - Button for toggling
+ *   the visibility of a document properties overlay.
+ */
 
-  initialize: function secondaryToolbarInitialize(options, eventBus) {
-    this.eventBus = eventBus;
+/**
+ * @class
+ */
+var SecondaryToolbar = (function SecondaryToolbarClosure() {
+  /**
+   * @constructs SecondaryToolbar
+   * @param {SecondaryToolbarOptions} options
+   * @param {EventBus} eventBus
+   */
+  function SecondaryToolbar(options, eventBus) {
     this.toolbar = options.toolbar;
-    this.buttonContainer = this.toolbar.firstElementChild;
-
-    // Define the toolbar buttons.
     this.toggleButton = options.toggleButton;
     this.presentationModeButton = options.presentationModeButton;
```

Comment:
I think that the `Button` suffix looks better, especially when comparing cases such as `this.print` with `this.printButton` and `this.download` with `this.downloadButton`. For me, the `Button` suffix makes it more clear that it is actually a reference to a button object, instead of an action like printing or downloading. I do see that this is a personal preference though.

To resolve the concern somewhat, I managed to get rid of the separate member variables for the buttons altogether by moving the button array to the constructor. This makes it look cleaner because we do not have to mention all the buttons twice anymore. I only had to pass in the hand tool button to the `_bindHandToolListener` method as that was the only method that actually used one of the button member variables.

Fun fact: after this change we have exactly `+178 −178` lines for this patch :-)